### PR TITLE
fix for issue 897, cosmic emulator omega_M missing massive neutrinos

### PR DESCRIPTION
**Pull request by patricialarsen**
Fix for https://github.com/LSSTDESC/CCL/issues/897

This changes two things 

- Alters the calculation of Omega_M used to call the emulator from omega_c+omega_b to omega_c+omega_b+omega_nu
- Alters the cosmology Omega_M values for the benchmark tests to the same definition

This should fix the issue without breaking the benchmark, and make the linear theory spectra more consistent with the other implementations when incorporating massive neutrinos.

### CODE REVIEW

**Review by elisachisari — COMMENTED** [review]
The changes you proposed makes sense: adding neutrinos to emu_par[0], which was incorrectly interpreted as not including neutrinos from Table 1 of https://arxiv.org/pdf/1705.03388.pdf. This requires changing the benchmark cosmologies as well, and I've checked this is done consistently. 
The only thing I can suggest is adding the check you did to discover this issue to the benchmarks. Would that be possible?

**Comment by patricialarsen** [comment]
Interesting suggestion, I did two checks to discover the issue - one was the same as the benchmark, but checking over the cosmologies for the input points and confirming the definition of omega_M, so that is now consistent with what is there. 

The other is comparing the low-k values against the halofit/linear power spectra with the same cosmology from CCL. At the emulation points at redshift 0 (restricting to cosmologies without a w=-1 crossing) you should see agreement within about 1%

image

So we could add a check that between k= 10^-3 and 10^-2 the power spectra should be consistent to within, say, 3% of the linear power spectrum at a selection of emulation points. At higher redshifts time evolving dark energy implementations can have differences at very linear scales, so this should be implemented at z=0. It would also catch any bugs in how you specify w0/wa.

**Comment by patricialarsen** [comment]
Note: new commit is just adding a line to the changelog

**Comment by patricialarsen** [comment]
I have added the new benchmark test for a selection of 3 emulator input cosmologies (taken from the Lawrence et al 2017 paper and written in benchmarks/data/emu_input_cosmologies.txt), as suggested above. This is added within test_emu.py and called test_emu_lin. It compares the emulator power spectrum between k=10^-3 and k=10^-2 to the linear power spectrum and enforces a <3% agreement. 

This would catch the issue above, and enforces agreement between the linear power spectrum and the emulator. These benchmarks all pass when I run the test. 

Further commit is fixing lint issues in the benchmark test.

**Comment by patricialarsen** [comment]
Note: update is changing the linear comparison to boltzmann_camb rather than class. The emulator was built off a modified camb implementation and so this is a closer comparison to the input.  You can see the difference in the fractional error for the model that was failing the integration test below. 

I suspect the difference is due to differences in the way time evolving dark energy is handled as this model has a high wa value. The emulator should be very consistent with the camb fluid dark energy implementation, while class by default uses the PPF approximations which can diverge at very low k (but have the benefit of allowing w=-1 crossings). If you look for instance at https://camb.info/ppf/ you can see the known issues 

camb_class

File: benchmarks/test_emu.py [hunk 1]
```diff
@@ -73,3 +73,47 @@ def test_emu(model):
...
+    a = 1
+    k = np.logspace(-3, -2, 50)
+
+    # Catch warning about neutrino linear growth
+    if (np.sum(mnu) > 0):
+        pk = ccl.pyutils.assert_warns(ccl.CCLWarning,
+                                      ccl.nonlin_matter_power,
+                                      cosmo, k, a)
+    else:
+        pk = ccl.nonlin_matter_power(cosmo, k, a)
```
Comment (elisachisari) [inline benchmarks/test_emu.py:111]: Small thing - should we use ccl.linear_matter_power here? Same above in 108

Comment (patricialarsen) [reply benchmarks/test_emu.py:111]: I'm not sure what you mean: these two lines are intended to get the non-linear matter power from the emulator for the comparison, and the linear_matter_power for the comparison is in line 114.

Comment (elisachisari) [reply benchmarks/test_emu.py:111]: Ah, ok, sorry I was thinking of the line 114 input indeed

**Review by elisachisari — APPROVED** [review]
Nice work by @patricialarsen catching this issue and fixing it. We iterated a bit on the changes and I think the PR is now ready to merge. The changelog has been updated and new benchmarks have been provided.

**Pull request merged**